### PR TITLE
[MM-52550] platform/web_conn: fix a possible data race issue

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We run a plugin/product hook once we create a `WebConn` 

https://github.com/mattermost/mattermost-server/blob/24f484b012a626f186655c240bec7820ea26c2d6/server/channels/app/platform/web_conn.go#L238-L241

but since we are running this hook on a separate goroutine, the value is re-written in the `ServeWebSocket` if the request action is ` model.WebsocketAuthenticationChallenge` because we get the userID from the session token this time. 

https://github.com/mattermost/mattermost-server/blob/e300cb942adaea3c32d5e9f9ed71a7eae93392c2/server/channels/app/platform/websocket_router.go#L57

To prevent some inconsistencies and race, we can just copy values for the initial connect.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52550

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by agarciamontoro** [comment]
Why does the `WebsocketAuthenticationChallenge` action change the user ID? Is that needed?

File: server/channels/app/platform/web_conn.go [hunk 1]
```diff
@@ -232,9 +232,11 @@ func (ps *PlatformService) NewWebConn(cfg *WebConnConfig, suite SuiteIFace, runn
 	wc.SetSessionExpiresAt(cfg.Session.ExpiresAt)
 	wc.SetConnectionID(cfg.ConnectionID)
 
-	wc.Platform.Go(func() {
-		wc.HookRunner.RunMultiHook(func(hooks plugin.Hooks) bool {
-			hooks.OnWebSocketConnect(wc.GetConnectionID(), wc.UserId)
+	connID := wc.GetConnectionID()
```
Comment (agnivade) [inline server/channels/app/platform/web_conn.go:235]: connectionID is already an atomic, so I don't think we need that. The race is only happening on accessing `wc.UserId`.

However, `wc.UserId` is being accessed in a lot of other `platform.Go` methods as well. Let's change those as well for completeness.

**Comment by agnivade** [comment]
> Why does the WebsocketAuthenticationChallenge action change the user ID? Is that needed?

Yes, in case it's authenticated via the auth challenge, it's needed. How else would you set the userID?

File: server/channels/app/platform/web_conn.go (hunk 1 above)
Comment (isacikgoz) [reply server/channels/app/platform/web_conn.go:235]: Makes sense 👍

**Comment by agarciamontoro** [comment]
> Yes, in case it's authenticated via the auth challenge, it's needed. How else would you set the userID? (Unless you mean to get that from the session every time, which is also a solution)

I think I lack context for how all this is architected, I was just wondering why the userID would change in the middle of the connection (thus generating this data race).

**Review by agnivade — APPROVED** [review]
Thanks!

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Comment by isacikgoz** [comment]
> I was just wondering why the userID would change in the middle of the connection (thus generating this data race).

@agarciamontoro  It's because auth challenge gets the user id from token after validating it, I think it was expected. And we're mering this before your approval because we are going to run some tests, feel free to raise any issue if you have, I'll address them in a separate PR.

**Pull request merged**